components/org.wso2.carbon.deployment.engine/src/main/java/org/wso2/carbon/deployment/engine/internal/DeploymentEngineListenerComponent.java

```diff
@@ -171,15 +174,41 @@ protected void unregisterCarbonRuntime(CarbonRuntime carbonRuntime) {
         DataHolder.getInstance().setCarbonRuntime(null);
     }
 
+    /**
+     * Get the ConfigProvider service.
+     * This is the bind method that gets called for ConfigProvider service registration that satisfy the policy.
+     *
+     * @param configProvider the ConfigProvider service that is registered as a service.
+     */
+    @Reference(
+            name = "carbon.config.provider",
+            service = ConfigProvider.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unregisterConfigProvider"
+    )
+    protected void registerConfigProvider(ConfigProvider configProvider) {
+        DataHolder.getInstance().setConfigProvider(configProvider);
+    }
+
+    /**
+     * This is the unbind method for the above reference that gets called for ConfigProvider instance un-registrations.
+     *
+     * @param configProvider the ConfigProvider service that get unregistered.
+     */
+    protected void unregisterConfigProvider(ConfigProvider configProvider) {
+        DataHolder.getInstance().setConfigProvider(null);
+    }
+
     @Override
     public void onAllRequiredCapabilitiesAvailable() {
         try {
             // Initialize deployment engine and scan it
-            String carbonRepositoryLocation = DataHolder.getInstance().getCarbonRuntime().getConfiguration().
-                    getDeploymentConfig().getRepositoryLocation();
+            DeploymentConfiguration deploymentConfiguration = DataHolder.getInstance().getConfigProvider()
+                    .getConfigurationObject(DeploymentConfiguration.class);
```

Comment:
Shall we reuse this deploymentConfiguration object within the component. Use it in other places where we need to read configuration.